robot/src/main/java/us/ilite/robot/driverinput/DriverInput.java

```diff
@@ -11,46 +13,51 @@
 import us.ilite.common.types.input.EInputScale;
 import us.ilite.common.types.input.ELogitech310;
 import us.ilite.robot.Data;
-import us.ilite.robot.commands.ICommand;
+import us.ilite.robot.commands.Delay;
 import us.ilite.robot.modules.Drive;
 import us.ilite.robot.modules.DriveMessage;
 import us.ilite.robot.modules.Module;
-
-import java.util.LinkedList;
-import java.util.Queue;
+import us.ilite.robot.modules.Superstructure;
 
 public class DriverInput extends Module {
 
     private static final double DRIVER_SUB_WARP_AXIS_THRESHOLD = 0.5;
+    private ILog mLog = Logger.createLog(DriverInput.class);
 
     protected final Drive driveTrain;
+    protected final Superstructure mSuperstructure;
+
+    private boolean mDriverStartedCommands;
 
-    private Queue<ICommand> desiredCommandQueue;
-    private boolean lastRunCommandQueue;
-    private boolean runCommandQueue;
     private Joystick mDriverJoystick;
     private Joystick mOperatorJoystick;
 
     private Codex<Double, ELogitech310> mDriverInputCodex, mOperatorInputCodex;
 
     private Data mData;
 
-    public DriverInput(Drive pDrivetrain, Data pData) {
+    public DriverInput(Drive pDrivetrain, Superstructure pSuperstructure, Data pData, boolean pSimulated) {
         this.driveTrain = pDrivetrain;
+        this.mSuperstructure = pSuperstructure;
         this.mData = pData;
-        this.desiredCommandQueue = new LinkedList<>();
         this.mDriverInputCodex = mData.driverinput;
         this.mOperatorInputCodex = mData.operatorinput;
-        this.mDriverJoystick = new Joystick(0);
-        this.mOperatorJoystick = new Joystick(1);
+        if(pSimulated) {
+            // Use a different joystick library?
```

Comment:
Are things going to break if the mDriverJoystick and mOperatorJoystick are null? Maybe use Optional?

Reply:
They aren't null; they are codexes instantiated at robot init.